### DISCUSSION TITLE
Getting started for self-hosted

### DISCUSSION THREAD

**Discussion by hinogi** [graphql-hive/console#296] [Help] [answered]
Could you add a section for setting up graphql hive in a self hosted env?

**Comment by KennethWussmann** [comment]
I would also appreciate a more self-hosting friendly setup. 

It appears that when you want to deploy your own schema registry you will get an entire SaaS platform designed for customers, involving billing, cloud infrastructure, Kubernetes, etc. 
While the more realistic intention to self-host is to host the GraphQL for your company or project. 
I would love to see core functionality being easier to host for example with Docker.

**Comment by kamilkisiela** [comment]
We're going to share docker images soon.

> It appears that when you want to deploy your own schema registry you will get an entire SaaS platform designed for customers, involving billing, cloud infrastructure, Kubernetes, etc.
While the more realistic intention to self-host is to host the GraphQL for your company or project.

True and not true. The billing is optional. The Kubernetes (pulumi setup) is also optional. We use it for deployment and it's public only because we wanted to show a real-world and working deployment setup for others. Only the secrets are secret.

**Reply by kamilkisiela** [reply]
#312 #311

**Comment by kamilkisiela** [comment, marked as answer]
https://docs.graphql-hive.com/self-hosting/get-started
